application/src/main/java/uk/nhs/hee/tis/revalidation/service/DoctorsForDBService.java

```diff
@@ -162,6 +162,25 @@ public void updateDoctorConnection(final ConnectionMessageDto message) {
     }
   }
 
+  public void disconnectDoctorsFromDb(final DoctorsForDbCollectedEvent doctorsForDbCollectedEvent) {
+    final String designatedBodyCode = doctorsForDbCollectedEvent.designatedBodyCode();
+    final LocalDateTime requestDateTime = doctorsForDbCollectedEvent.requestDateTime();//00:24
+    List<DoctorsForDB> doctorsForDBList = doctorsRepository.findByDesignatedBodyCode(
```

Comment:
would be more efficient to filter at the repository, `findByDesignatedBodyCodeAndBeforeGmcLastUpdatedDateTime` - then you don't really need the "handleDisconnections" method (much easier to test)